src/gui/sdlmain.cpp

```diff
@@ -1112,108 +1106,82 @@ static void schedule_synced([[maybe_unused]] const uint32_t event_id = 0)
-	};
-
-	auto configure_vfr_mode = [&]() {
-		set_vfr_dupe_countdown_from_rate(dos_rate);
-		const auto bench_rate = get_vsync_preference().benchmarked_rate;
-		const auto lesser_rate = std::min(host_rate, dos_rate);
-		save_rate_to_frame_period(lesser_rate);
-		const auto is_fast_enough = sdl.vsync.current != VSYNC_STATE::ON &&
-		                            atleast_as_fast(bench_rate, dos_rate);
-		const auto wont_hit_vsync_wall = sdl.vsync.current != VSYNC_STATE::OFF &&
-		                                 atleast_as_fast(host_rate, dos_rate);
-		return (is_fast_enough || wont_hit_vsync_wall)
-		               ? FRAME_MODE::VFR
-		               : FRAME_MODE::THROTTLED_VFR;
-	};
-
-	const bool wants_vsync = sdl.vsync.current == VSYNC_STATE::ON ||
-	                         get_vsync_preference().requested == VSYNC_STATE::ON;
-
 	// to be set below
 	auto mode = FRAME_MODE::UNSET;
 
-	// Text modes always get VFR
-	const bool in_text_mode = CurMode->type & M_TEXT_MODES;
-	if (in_text_mode) {
-		mode = FRAME_MODE::VFR;
+	// Manual CFR or VFR modes
+	if (sdl.frame.desired_mode == FRAME_MODE::CFR ||
+	    sdl.frame.desired_mode == FRAME_MODE::VFR) {
+		mode = sdl.frame.desired_mode;
+
+		// Frames will be presented at the DOS rate (if CFR) or
+		// duplicated at a low-rate (if VFR), with the pacer acting to
+		// inform the user if the host can't handle those rates.
 		save_rate_to_frame_period(dos_rate);
-		set_vfr_dupe_countdown_from_rate(dos_rate);
+		render_pacer->SetTimeout(sdl.vsync.skip_us);
 	}
-
-	// Manual full CFR
-	else if (sdl.frame.desired_mode == FRAME_MODE::CFR) {
-		if (configure_cfr_mode() != FRAME_MODE::CFR && wants_vsync) {
-			LOG_WARNING("SDL: CFR performance warning: the DOS rate of %2.5g"
-			            " Hz exceeds the host's %2.5g Hz vsynced rate",
-			            dos_rate, host_rate);
+	// Automatic CFR or VFR modes
+	else {
+		// What's the effective supported rate?
+		const auto bench_rate = get_vsync_preference().benchmarked_rate;
+		const double supported_rate = (bench_rate < host_rate ? bench_rate
+		                                                      : host_rate);
+		const bool can_support_dos_rate = (supported_rate > dos_rate);
+
+		// Is vsync on or requested?
+		const bool wants_vsync = sdl.vsync.current == VSYNC_STATE::ON ||
+		                         get_vsync_preference().requested ==
+		                                 VSYNC_STATE::ON;
+
+		const bool is_display_vrr = (host_rate >= REFRESH_RATE_HOST_VRR_MIN);
+
+		const bool in_graphics_mode = (CurMode->type & M_TEXT_MODES) == 0;
+
+		// vsynced VRR displays are typically doing frame-interpolation
+		// and need a constant rates to lock into the content. We also
+		// consider text-mode an exception to this because it's more
+		// often desirable to run text modes at relaxed, slow rates --
+		// so those fall through to VFR. Rare cases where user really
+		// care about this can always set CFR mode manually.
+		//
```

Comment:
Objection, did we not say we'll remove the special-casing of text modes entirely? Can we not do this to remove even the possibility of any graphics vs text mode complications? For example, I like to use _really_ fast keyrepeat rates in text modes, which would force high FPSes but only sporadically when I'm moving the cursor. Maybe not an issue, but I guess I don't like the concept of treating text modes any differently.

Just wanna to remove the continuous back and forth on fixing this for text modes. I propose to treat output as output—after all, the monitor does not "know" whether you're using a text mode, or a graphics mode, or maybe a graphics mode for text output, or a text mode for some basic graphics in a game... What about edge cases? Fast paced text mode game, or a graphics mode text editor with _no_ blinking cursor where you stare at screen? I think this differentation between graphics and text modes assumes too much...

Reply:
Was able to remove it again.